helpers/wrapper/migrations/2_deploy_contracts.js

```diff
@@ -7,7 +8,8 @@ module.exports = deployer => {
   deployer.deploy(Types)
   deployer.link(Types, Swap)
   deployer.deploy(WETH9)
-  deployer
-    .deploy(Swap)
+  deployer.deploy(TokenRegistry)
+    .then(() => TokenRegistry.deployed())
+    .then(() => deployer.deploy(Swap, TokenRegistry.address))
```

Comment:
why does this need to be in a `.then`? Can't it be done as a new `deployer.deploy` line like the deployments on lines 8-11?

Reply:
A tokenRegistry address needs to be accessible similar to how swap is needed for wrapper. deployer.deploy returns a promise so just having a list of promises would not be sufficient, I chained them so they were getting each others results.

Reply:
Yeah makes sense. You can fix it using `await` on each line but isn't a problem as is